lib/Board.rb

```diff
@@ -0,0 +1,95 @@
+class Board
```

Comment:
It appears Board class is doing so many things. Here are things to ask when designing this class (as well as other classes):

1. What data should Board class be holding? 
2. In relation to this data, what **single responsibility** should Board class have? The methods then that you write for this class should only be related to this single responsibility. Let's keep our classes lean and free from other methods that are better suited for other classes. And also let's make sure our Classes are not doing any UI related logic.

Reply:
Your board still has lots of methods in it. Board isn't responsible for determining the winner (it's the game's responsibility). The board should only be concerned with taking a player's choice and updating the board array as well as determining if it still has empty squares.